db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/AggregateNode.java

```diff
@@ -434,6 +434,27 @@ public ValueNode bindExpression(FromList fromList,
 		         operand = castNode.bindExpression( fromList, subqueryList, aggregateVector );
 		     }
 		}
+		else if (isWindowFunction() &&
+                 operand != null    &&
+                 operand.getTypeId().getTypeFormatId() !=
+                 resultType.getTypeId().getTypeFormatId()) {
+            // For now, the receiver data type picked by
+            // getAggregator() must match the data type of the
+            // operand when processing window functions
+            // to avoid ClassCastExceptions for NativeSparkDataSets.
+            // For non-windowed aggregations, we process the aggregation using
+            // the original data type of the operand for performance.  If it overflows,
+            // a new aggregator that uses the final result type is allocated.
+            // When there is no overflow, a final CAST is applied in
+            // methoc SpliceGeneratorAggregator.finish.
+            operand =
```

Comment:
Good catch.  We likely only want to do the CAST to avoid overflow when we're doing a SUM or AVG function, which will have a compatible data type with the operand.  I will add a condition to only do the cast for SUM or AVG:
`uad instanceof SumAvgAggregateDefinition`